pulsator4droid/src/main/java/pl/bclogic/pulsator4droid/library/circle/CirclePulsatorLayout.java

```diff
@@ -0,0 +1,392 @@
+                // disable the onDraw method (masking)
+                setWillNotDraw(true);
+                mCircularMaskRadius = RADIUS_NONE;
+                mMask = null;
+            } else {
+                // enable the onDraw method (masking)
+                setWillNotDraw(false);
+                mCircularMaskRadius = radius;
+                // mask is being initialized on demand during #onDraw(Canvas) method (now enabled)
+                // to make sure there has already been #onMeasure(int, int) pass
+            }
+        }
+    }
+
+    /**
+     * Build pulse views and animators.
+     */
+    private void build() {
+        // create views and animators
+        LayoutParams layoutParams = new LayoutParams(
+                LayoutParams.MATCH_PARENT,
+                LayoutParams.MATCH_PARENT);
+
+        int repeatCount = (mRepeat == INFINITE) ? ObjectAnimator.INFINITE : mRepeat;
+
+        mAnimators = new ArrayList<>(3 * mCount);
+        for (int index = 0; index < mCount; index++) {
+            // setup view
+            PulseView pulseView = new PulseView(getContext());
+            pulseView.setScaleX(0);
+            pulseView.setScaleY(0);
+            pulseView.setAlpha(1);
+
+            addView(pulseView, index, layoutParams);
+            mViews.add(pulseView);
+
+            long delay = index * mDuration / mCount;
+
+            // setup animators
+            ObjectAnimator scaleXAnimator = ObjectAnimator.ofFloat(pulseView, "ScaleX", 0f, 1f);
+            scaleXAnimator.setStartDelay(delay);
+            mAnimators.add(scaleXAnimator);
+
+            ObjectAnimator scaleYAnimator = ObjectAnimator.ofFloat(pulseView, "ScaleY", 0f, 1f);
+            scaleYAnimator.setStartDelay(delay);
+            mAnimators.add(scaleYAnimator);
+
+            ObjectAnimator alphaAnimator = ObjectAnimator.ofFloat(pulseView, "Alpha", 1f, 0f);
+            alphaAnimator.setStartDelay(delay);
+            mAnimators.add(alphaAnimator);
+        }
+
+        for (Animator animator : mAnimators) {
+            ObjectAnimator objectAnimator = (ObjectAnimator) animator;
+            objectAnimator.setRepeatCount(repeatCount);
+            objectAnimator.setRepeatMode(ObjectAnimator.RESTART);
+            objectAnimator.setInterpolator(createInterpolator(mInterpolator));
+            objectAnimator.setDuration(mDuration);
+        }
+
+        if (mAnimators.isEmpty()) {
+            mAnimators = null;
+        } else {
+            mAnimators.get(0).addListener(mAnimatorStartListener);
+            mAnimators.get(mAnimators.size() - 1).addListener(mAnimatorEndListener);
+        }
+    }
+
+    private class PulseView extends View {
+
+        public PulseView(Context context) {
+            super(context);
+        }
+
+        @Override
+        protected void onDraw(Canvas canvas) {
+            mPulseShape.draw(canvas);
+        }
+
+    }
+
+    private class AnimatorSimpleListener implements Animator.AnimatorListener {
```

Comment:
This seems like a convenience class and it quite messes up the class code. Also it is instantiated in fields after its declaration. The fields should be at the beginning of the class.

Reply:
Actually the class seems like a "Util" that could be used everywhere, so maybe it would deserve its own class

Reply:
Again, this is just moved original code of the library. If we want to ever apply any changes/PRs from the original library, the more changes and refactoring we do in our fork, the more difficult the process will be.